plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/attribute-options/frontend.ts

```diff
@@ -1,9 +1,152 @@
 /**
  * External dependencies
  */
-import { store } from '@wordpress/interactivity';
+import { store, getContext, getElement } from '@wordpress/interactivity';
 
-store( 'woocommerce/add-to-cart-with-options', {
-	state: {},
-	actions: {},
-} );
+type Option = {
+	value: string;
+	label: string;
+	isSelected: boolean;
+};
+
+type Context = {
+	selected: string | null;
+	option: Option;
+	options: Option[];
+};
+
+type PillsContext = Context & {
+	tabIndex?: number;
+	selected?: string;
+	focused?: string;
+};
+
+store(
+	'woocommerce/add-to-cart-with-options-variation-selector-attribute-options__pills',
+	{
+		state: {},
+		actions: {
+			handleClick() {
+				const context = getContext< PillsContext >();
+				if ( context.selected === context.option.value ) {
+					context.selected = '';
+				} else {
+					context.selected = context.option.value;
+				}
+				context.focused = context.option.value;
+			},
+			handleKeyDown( event: KeyboardEvent ) {
+				const context = getContext< PillsContext >();
+
+				let flag = false;
```

Comment:
What do you think about adding a more description name to this variable? Something like `keyWasProcessed` or `shouldPreventEvent`.

Reply:
done